### PR TITLE
EMP-2622, restrict all permissions under elasticfilesystem:*

### PR DESCRIPTION
**Pull request by Rohitrajak1807**
Ticket link: https://platform9.atlassian.net/browse/EMP-2622

Testing:
EMP create and delete works

### CODE REVIEW

File: emp/emp_role_cftemplate.yaml [hunk 1]
```diff
@@ -301,23 +301,43 @@ Resources:
...
+            Condition:
+              StringEquals:
+                aws:RequestTag/emp.pf9.io: owned
+              StringLike:
+                aws:RequestTag/emp.pf9.io/baremetalpool: '*'
+                aws:RequestTag/emp.pf9.io/namespace: '*'
+          - Action:
+              - elasticfilesystem:DescribeFileSystems
               - elasticfilesystem:CreateMountTarget
-              - elasticfilesystem:CreateTags
```
Comment (AnirudhPokala) [inline emp/emp_role_cftemplate.yaml:308]: We are going to add tags to EFS from controllers I believe? Doesn't it require this permission of createTags for an EFS?

Comment (Rohitrajak1807) [reply emp/emp_role_cftemplate.yaml:308]: It is there

Comment (Rohitrajak1807) [reply emp/emp_role_cftemplate.yaml:308]: See: https://github.com/platform9/support-locker/pull/54/commits/e31ca14f4b395aa55601ef075d1294c157478382

Comment (AnirudhPokala) [reply emp/emp_role_cftemplate.yaml:308]: As discussed, so this permission is not needed specifically, but we do have the permissions to add tags cc: @Rohitrajak1807

**Review by AnirudhPokala — APPROVED** [review]

File: emp/emp_role_cftemplate.yaml [hunk 2]
```diff
@@ -301,23 +301,43 @@ Resources:
...
+          - Action:
               - elasticfilesystem:TagResource
-              - elasticfilesystem:UntagResource
-            Resource:
-              - '*'
             Effect: Allow
+            Resource: '*'
+            Condition:
+              StringEquals:
+                elasticfilesystem:CreateAction: 'CreateFileSystem'
```
Comment (Rohitrajak1807) [inline emp/emp_role_cftemplate.yaml:340]: .

File: emp/emp_role_cftemplate.yaml (hunk 1 above)
Comment (Rohitrajak1807) [reply emp/emp_role_cftemplate.yaml:308]: yes. it's here: https://github.com/platform9/support-locker/pull/54/files#r1773132869

**Pull request merged**